src/idl_parser.cpp

```diff
@@ -1944,6 +1956,19 @@ CheckedError Parser::ParseSingleValue(const std::string *name, Value &e,
     // Integer token can init any scalar (integer of float).
     FORCE_ECHECK(kTokenIntegerConstant, IsScalar(in_type), BASE_TYPE_INT);
   }
+  // Match empty vectors for default-empty-vectors.
+  if (!match && IsVector(e.type) && token_ == '[') {
+    const char *c = cursor_;
+    // Peek next token.
+    while (*c == ' ' || *c == '\t' || *c == '\n') c++;
```

Comment:
I don't think we want special purpose lexing outside the lexer :)
Why not just consume the `[` and check for the `]` normally? it's not like `[` starts any other legitimate constructs in this context?